### PR TITLE
an AWS janitor for boskos

### PR DESCRIPTION
**Pull request by liztio**
in the style of the existing janitor.go and janitor.py

### CODE REVIEW

**Comment by liztio** [comment]
Part of #10693

**Comment by krzyzacy** [comment]
we probably can also bazel-fy boskos like prow :-)

also let me know if this is ready for review.

**Comment by liztio** [comment]
To build the docker image: `$ bazel build //maintenance/aws-janitor/cmd/aws-janitor-boskos:aws-janitor-boskos-image`

**Comment by rdodev** [comment]
The docker image produced by that command above isn't a valid docker image. It complains about lacking json manifest.

**Comment by liztio** [comment]
@rdodev huh, what commands are you running? It works for me:
```
$ docker run bazel/maintenance/aws-janitor/cmd/aws-janitor-boskos:aws-janitor-boskos-image 
{"error":"Couldn't retrieve resources from Boskos: Post http://boskos/acquire?type=aws-account\u0026state=dirty\u0026dest=cleaning\u0026owner=AWSJanitor: dial tcp: lookup boskos: no such host","level":"error","msg":"Janitor failure","time":"2019-02-08T22:17:15Z"}
```

**Comment by liztio** [comment]
I tested this on a Heptio Quickstart, and once I added the NAT gateway cleaner it worked great. We may need to add a few more AWS resources at some point, but that can come later.

**Comment by krzyzacy** [comment]
/approve
/hold
fix lint errors :-)

File: maintenance/aws-janitor/resources/nat_gateway.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+type natGateway struct {
+	Account string
+	Region  string
+	ID      string
+}
+
+func (ng natGateway) ARN() string {
+	return fmt.Sprintf("arn:aws-cn:ec2:%s:%s:natgateway/%s", ng.Region, ng.Account, ng.ID)
```
Comment (justinsb) [inline maintenance/aws-janitor/resources/nat_gateway.go]: Presumably not aws-cn (not that it matters!)

Comment (liztio) [reply maintenance/aws-janitor/resources/nat_gateway.go]: oh whoops! I'll grab that in my next PR

**Pull request merged**